core/src/main/java/com/linecorp/armeria/server/ServiceRequestContext.java

```diff
@@ -258,6 +260,28 @@ default String pathParam(String name) {
         return pathParams().get(name);
     }
 
+    /**
+     * Returns the query parameters mapped by the {@link Route} associated with the {@link Service}
+     * that is handling the current {@link Request}.
+     */
+    QueryParams queryParams();
+
+    /**
+     * Returns the value of the specified query parameter.
+     */
+    @Nullable
+    default String queryParam(String name) {
+        return queryParams().get(name);
+    }
+
+    /**
+     * Returns the values of the specified query parameter.
+     */
+    @Nullable
+    default List<String> queryParams(String name) {
+        return queryParams().getAll(name);
+    }
```

Comment:
I found that `QueryParams` also support `List<String> getAll(name)` so added this method too!

WDYT? If this method is unnecessary, then I'll remove it!

Reply:
I think it's okay. 😄